moveit_ros/visualization/rviz_plugin_render_tools/src/trajectory_visualization.cpp

```diff
@@ -202,7 +202,7 @@ void TrajectoryVisualization::reset()
 void TrajectoryVisualization::clearTrajectoryTrail()
 {
   for (std::size_t i = 0; i < trajectory_trail_.size(); ++i)
-    delete trajectory_trail_[i];
+    trajectory_trail_[i].reset();
```

Comment:
There's no need for this loop anymore. shared_ptr are cleared implicitly in the next line.